packages/core/src/sync-historical/service.ts

```diff
@@ -898,17 +952,33 @@ export class HistoricalSyncService extends Emittery<HistoricalSyncEvents> {
 
   private _eth_getLogs = async (
     options: LogFilterCriteria & { fromBlock: Hex; toBlock: Hex },
+    signal: AbortSignal,
   ) => {
     const logs: RpcLog[] = [];
 
     let error: (Partial<RpcError> & { name: string }) | null = null;
 
     const stopClock = startClock();
     try {
-      return await this.network.client.request({
-        method: "eth_getLogs",
-        params: [options],
+      const rawRequest = await this.network.request({
+        body: {
+          method: "eth_getLogs",
+          params: [options],
+        },
+        fetchOptions: { signal },
       });
+
+      if (rawRequest.error)
+        throw new RpcRequestError({
```

Comment:
I'm a little worried about this, because we take advantage of viem error formats in the _eth_getLogs helper. This probably results in different errors.

Reply:
I think this should be fine because this error is the exact same as the one viem is throwing

Reply:
Ok, got it.